packages/block-editor/src/components/block-list/use-block-props/use-selected-block-event-handlers.js

```diff
@@ -134,31 +117,136 @@ export function useEventHandlers( { clientId, isSelected } ) {
 				ownerDocument.body.appendChild( dragElement );
 				event.dataTransfer.setDragImage( dragElement, 0, 0 );
 
-				let offset = { x: 0, y: 0 };
+				const rect = node.getBoundingClientRect();
+
+				const clone = node.cloneNode( true );
+				clone.style.visibility = 'hidden';
+				// Maybe remove the clone now that it's relative?
+				clone.style.display = 'none';
+
+				// Remove the id and leave it on the clone so that drop target
+				// calculations are correct.
+				const id = node.id;
+				node.id = null;
+
+				let _scale = 1;
 
-				if ( document !== ownerDocument ) {
-					const frame = defaultView.frameElement;
-					if ( frame ) {
-						const rect = frame.getBoundingClientRect();
-						offset = { x: rect.left, y: rect.top };
+				let parentElement = node;
+
+				while ( ( parentElement = parentElement.parentElement ) ) {
+					const { scale } =
+						defaultView.getComputedStyle( parentElement );
+					if ( scale && scale !== 'none' ) {
+						_scale = parseFloat( scale );
+						break;
 					}
 				}
 
-				// chip handle offset
-				offset.x -= 58;
+				let bgColor = 'transparent';
 
-				function over( e ) {
-					domNode.style.transform = `translate( ${
-						e.clientX + offset.x
-					}px, ${ e.clientY + offset.y }px )`;
+				parentElement = node;
+
+				while ( ( parentElement = parentElement.parentElement ) ) {
+					const { backgroundColor } =
+						defaultView.getComputedStyle( parentElement );
+					if (
+						backgroundColor &&
+						backgroundColor !== 'transparent' &&
+						backgroundColor !== 'rgba(0, 0, 0, 0)'
+					) {
+						bgColor = backgroundColor;
+						break;
+					}
+				}
+
+				const inverted = 1 / _scale;
+
+				node.after( clone );
+
+				const originalNodeProperties = {};
+				for ( const property of [
+					'transform',
+					'transformOrigin',
+					'transition',
+					'zIndex',
+					'position',
+					'top',
+					'left',
+					'pointerEvents',
+					'opacity',
+					'backgroundColor',
+				] ) {
+					originalNodeProperties[ property ] = node.style[ property ];
 				}
 
-				over( event );
+				// Get scroll position.
+				const originScrollTop = defaultView.scrollY;
+				const originScrollLeft = defaultView.scrollX;
+				const originClientX = event.clientX;
+				const originClientY = event.clientY;
+
+				// We can't use position fixed because it will behave different
+				// if the html element is is scaled or transformed (position
```

Comment:
Minor typo, double `is` 
```suggestion
				// if the html element is scaled or transformed (position
```

Reply:
Fixed